VideoApp/VideoApp/Stores/Auth/PasscodeComponents.swift

```diff
@@ -18,10 +18,29 @@ import Foundation
 
 struct PasscodeComponents {
     let passcode: String
-    let appID: String
+    let appID: String?
+    let serverlessID: String
     
-    init(string: String) {
+    init(string: String) throws {
         passcode = string
-        appID = String(string.dropFirst(6))
+        let shortPasscodeLength = 6
+        let appIDLength = 4
+        let serverlessIDMinLength = 4
+        let newFormatMinLength = shortPasscodeLength + appIDLength + serverlessIDMinLength
+        let oldFormatMinLength = shortPasscodeLength + serverlessIDMinLength
+
+        if string.count >= newFormatMinLength {
+            let appIDStartIndex = string.index(string.startIndex, offsetBy: shortPasscodeLength)
+            let appIDEndIndex = string.index(appIDStartIndex, offsetBy: appIDLength - 1)
+            appID = String(string[appIDStartIndex...appIDEndIndex])
+            let serverlessIDStartIndex = string.index(after: appIDEndIndex)
+            serverlessID = String(string[serverlessIDStartIndex...])
+        } else if string.count >= oldFormatMinLength {
+            appID = nil
+            let serverlessIDStartIndex = string.index(string.startIndex, offsetBy: shortPasscodeLength)
+            serverlessID = String(string[serverlessIDStartIndex...])
```

Comment:
Pretty unfortunate that parsing strings is this painful in Swift.

Reply:
Can you use regular expression matching and capturing to get what you need?

![regex](https://user-images.githubusercontent.com/25950/89664487-3f371780-d8a5-11ea-93cf-14805ea2537a.jpg)

Reply:
Haha! Yeah I suspect there may be some regex magic that could do this but I admit I'm not a regex wizard and as your meme suggests it may not be better. But if I get some time I will look into it.

Reply:
It's not a necessity. I was just wondering if there was something that in the end would make it simpler. But yeah, it can also make it.... much much worse. 😀